src/commandline/commandlineparser.cpp

```diff
@@ -72,6 +73,33 @@ void CommandLineParser::setEngine(QQmlEngine* engine)
     }
 }
 
+void CommandLineParser::loadDummyData(QQmlEngine *engine)
+{
+    const QFileInfo file = ProvidesSomething::self()->filePath().toLocalFile();
+    const QString pathToQml = file.canonicalPath();
+
+    const QDir dir(pathToQml + "/dummydata/", "*.qml");
+    QStringList list = dir.entryList();
+    for (int i = 0; i < list.size(); ++i) {
```

Comment:
Please use a [range-based](https://learn.microsoft.com/en-us/cpp/cpp/range-based-for-statement-cpp?view=msvc-170) for loop